geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -253,7 +253,8 @@ public void firePendingCallbacks() {
     }
 
     for (EntryEventImpl ee : getPendingCallbacks()) {
-      boolean isLastTransactionEvent = isConfigError || ee.equals(lastTransactionEvent);
+      boolean isLastTransactionEvent =
+          isConfigError || lastTransactionEvent == null || ee.equals(lastTransactionEvent);
```

Comment:
This is the 2nd instance of this same code which was also in TXCommitMessage.java. You might want to create a static utils method and have both of these call that method.